dask/dataframe/core.py

```diff
@@ -351,6 +353,57 @@ def __init__(self, dsk, name, meta, divisions):
         self._meta = meta
         self.divisions = tuple(divisions)
 
+        # Optionally cast object dtypes to `pyarrow` strings
+        if dask.config.get("dataframe.convert_string"):
+            try:
+                import pyarrow  # noqa: F401
+            except ImportError:
+                raise RuntimeError(
+                    "Using dask's `dataframe.convert_string` configuration "
+                    "option requires `pyarrow` to be installed."
+                )
+            if not PANDAS_GT_130:
+                raise RuntimeError(
+                    "Using dask's `dataframe.convert_string` configuration "
+                    "option requires pandas>=1.3.0 to be installed. "
+                    f"pandas={str(PANDAS_VERSION)} is currently using used."
+                )
+
+            from dask.dataframe._pyarrow import (
+                is_object_string_dataframe,
+                is_object_string_dtype,
+                is_object_string_index,
+                is_object_string_series,
+                to_pyarrow_string,
+            )
+
+            if (
+                is_object_string_dataframe(meta)
+                or is_object_string_series(meta)
+                or is_object_string_index(meta)
+            ):
```

Comment:
```suggestion
            if (
                isinstance(meta, (pd.DataFrame, pd.Series, pd.Index)) and (
                    is_object_string_dataframe(meta)
                    or is_object_string_series(meta)
                    or is_object_string_index(meta)
                )
            ):
```
This is not the only solution to the "cudf" problem, but it's a relatively simple one.

Reply:
Hrm, it looks like that punts on having `dataframe.convert_string` work with cudf-backed DataFrames. I'm curious what in the current implementation isn't working.

Reply:
The problem is just that a cudf string column will *seem* to be an object dtype, even though the underlying memory layout is already arrow-like:

```
In [11]: cudf.Series(["cat", "dog"]).dtype
Out[11]: dtype('O')
```

Therefore, you need to explicitly check that you are actually dealing with a pandas object, or you will end up in a recursive loop to convert object dtypes to pyarrow-string dtypes (even though they are essentially pyarrow-string dtypes in the beginning).

Reply:
Ah, I see. Thanks for the extra context. So are you saying `dataframe.convert_dtypes` should just ignore `cudf` things altogether? Is all text data in `cudf` already backed by `pyarrow` strings?

Reply:
For reference, @rjzamora and I just chatted offline about this. It turns out that while text data in `cudf` aren't backed by `pyarrow` dtypes, they are using a performant memory layout that gives users similar benefits to `pyarrow` strings. In fact, Rick even mentioned that doing `.astype("string[pyarrow]")` on a `cudf` object is just a no-op. Because of this, we decided to make `dataframe.convert_dtypes` a no-op for non-`pandas` DataFrames. This seems like the right call for now, but is also something we can easily revisit in the future, if needed. (Rick, feel free to fill in any details I may have missed)